component/phlare/push.v1.proto

```diff
@@ -0,0 +1,38 @@
+syntax = "proto3";
```

Comment:
We discussed this offline. Since the intent is to be able to share the scrape and push code between Grafana Agent and Phlare, _and_ I am concerned about forcing our module dependency hell on Phlare, we should be careful about how we do this. 

It's likely we're going to eventually have the scrape and push code in its own module for sharing. That module cannot import any Grafana Agent package (otherwise it introduces the problem mentioned above), so we will do the following:

1. Create an `api` package which contains the scrape manager and client code to push profiles to Phlare. This `api` package will eventually be its own module, but will remain in the base module for now. 
2. We will suggest that protobufs move to a shared location / yet another module that clients can import from  to avoid a circular dependency between Grafana Agent and Phlare, but this isn't a blocker for this PR.

Reply:
That sounds like more my problem. 

Grafana also has to generate the[ query side ](https://github.com/grafana/grafana/tree/0845ac2f53ca17cf698341b382af5262335e295c/pkg/tsdb/phlare/gen/querier/v1)and I've been copying code so far, but I think it's time to have a Phlare API module as you suggest where protobuf generation happens.

I'll investigate it.

Reply:
Trying this here https://github.com/grafana/phlare/pull/466